backend/services/office_hours/section.py

```diff
@@ -0,0 +1,787 @@
+        ticket_duration_times = [
+            (ticket.closed_at - ticket.called_at).total_seconds() / 60
+            for ticket in filtered_ticket_entities
+            if (ticket.closed_at and ticket.called_at)
+        ]
+
+        # Compute the average
+        avg_duration_time = (
+            sum(ticket_duration_times) / len(ticket_duration_times)
+            if ticket_duration_times
+            else 0.0
+        )
+
+        # Compute the standard deviation of duration times; handle if built in stdev raises Error
+        try:
+            std_dev_duration_time = statistics.stdev(ticket_duration_times)
+        except statistics.StatisticsError:
+            std_dev_duration_time = 0.0
+
+        # --- Round floats to two decimal places, construct the model and return it ---
+        return OfficeHoursSectionTrailingWeekData(
+            number_of_tickets=len(filtered_ticket_entities),
+            number_of_students=len(unique_ticket_creators),
+            average_wait_time=round(avg_wait_time, 2),
+            standard_deviation_wait_time=round(std_dev_wait_time, 2),
+            average_ticket_duration=round(avg_duration_time, 2),
+            standard_deviation_ticket_duration=round(std_dev_duration_time, 2),
+        )
+
+    def get_oh_section_members(
+        self, subject: User, oh_section: OfficeHoursSectionDetails
+    ) -> list[SectionMember]:
+        """Retrieves all of the subject's called office hours tickets in a section from the table.
+        Args:
+            subject: a valid User model representing the currently logged in User
+            oh_section: the OfficeHoursSectionDetails to query by.
+        Returns:
+            list[SectionMemberDetails]: List of all `SectionMemberDetails` in an OHsection
+        """
+
+        # Select OH Section by id
+        query = select(OfficeHoursSectionEntity).where(
+            OfficeHoursSectionEntity.id == oh_section.id
+        )
+
+        oh_section_entity = self._session.scalars(query).one_or_none()
+
+        if oh_section_entity is None:
+            raise ResourceNotFoundException(
+                f"Unable to find section with id {oh_section.id}"
+            )
+
+        # PERMISSIONS
+
+        # Throws exception if user is not a member
+        section_member_entity = self._check_user_section_membership(
+            subject.id, oh_section_entity.id
+        )
+
+        # Raises PermissionError if user trying to fetch all members is a Student
+        if section_member_entity.member_role == RosterRole.STUDENT:
+            raise PermissionError(
+                f"Section Member is a Student. User Does Not Have Permission to Get Section Members."
+            )
+
+        # Get the members that are linked to the academic sections which are linked to the OH section
+        member_entities = [
+            member
+            for section in oh_section_entity.sections
+            for member in section.members
+        ]
```

Comment:
When the app scales, this might be a little slow / inefficient since there are two nested loops - is there a way to bake this into the query itself with joins?

Reply:
Good point! We have cleaned this up a bit by fetching the academic section, then accessing the members directly from there.